Changed PostgreSQL to 14 (#1958)
ibexa/documentation-developer: docs/getting_started/requirements.md

The minimal setup requires PHP, MySQL/MariaDB, Apache/Nginx, Node.js and `yarn`.
Recommendation for production setups is to use Varnish/Fastly, Redis/Memcached, NFS/EFS/S3 and Solr/Elasticsearch in a [clustered setup](clustering.md).

!!! caution "Recommended versions"

>     Using the latest listed version of each product or component is always recommended. Review all the recommended versions carefully. If you see a "+" next to the product version, it means that we recommend this version or higher within the same major release. For example, "Nginx 1.18+" means any 1.x version higher or equal to 1.18, but not 2.x.

Reviewer: Maybe we could add to this text something like: "Always use a version that receives security updates, either by the vendor themselves or by a trusted 3rd party like the distribution vendor."

!!! caution "Recommended versions"

    Review all the recommended versions carefully. If you see a "+" next to the product version, it means that we recommend this version or higher within the same major release. For example, "Nginx 1.18+" means any 1.x version higher or equal to 1.18, but not 2.x.

>     Using the latest listed version of each product or component is recommended. Always use a version that receives security updates, either by the vendor themselves or by a trusted third party for example, the distribution vendor.

Reviewer: Good reshuffle of the text. 👍  There should be a comma after "party". Also I prefer "like" or "such as", to reduce the number of commas and because there aren't many kinds of trusted 3rd parties here.
```suggestion
    Using the latest listed version of each product or component is recommended. Always use a version that receives security updates, either by the vendor themselves or by a trusted third party, such as the distribution vendor.
```

- MariaDB 10.3+
- MySQL 8.0

> - PostgreSQL 14

Reviewer 2: As mentioned internally, I think it would be safer for 3.3 to explicitly recommend it instead of requiring (unless this section is all about recommendation?). Do we intentionally skip `+` sign?

> - PostgreSQL 14

Reviewer: I'm not too happy about this 😑 For one, we might want to wait for the resolution of https://github.com/ezsystems/ezpublish-kernel/pull/3152 even if it isn't the same thing.

So we say "we recommend this version or higher within the same major release". That's problematic here since postgres mainly uses major versions. If we truly do recommend 14 or higher then sure, fine. But do we? I thought we just recommend currently supported postgres versions, which would be `11 - 15`. I think I would prefer that, once it is QA-ed 😉

Reviewer 3: I'm staring to think we should drop the "+" sign all together and replace it with lists explicitly stating (major) versions.

Reviewer: ⬆️ 🔥  You have a point.
